spec/draft-barnes-cfrg-4q.md

```diff
@@ -702,133 +717,137 @@ This document has no IANA actions.
-implementations.  Side-channel analysis is a constantly moving field,
-and implementers must be extremely careful to ensure that
-operations do not leak any secret information. Using
-ephemeral private scalars for each operation (ideally, limiting the 
-use of each private scalar to one single operation) is an effective measure to
-reduce the impact of side-channel attacks. However, this might not be
-possible for many applications of Diffie-Hellman key agreement.
-
-In the future quantum computers will render the discrete logarithm
-problem easy on all abelian groups through Schor's algorithm. Data
-intended to remain confidential for significantly extended periods of
-time SHOULD NOT be protected with any elliptic curve, finite field,
-or factorization based primitive.
+The best known algorithms for the computation of discrete logarithms on Curve4Q
+are parallel versions of the Pollard rho algorithm in [Distinguished]. On
+Curve4Q these attacks take on the order of 2^123 group operations to compute a
+single discrete logarithm. The additional endomorphisms have large order, and so
+cannot be used to accelerate generic attacks. Quadratic fields are not affected
+by any of the index calculus attacks used over larger extension fields.
+
+Implementations MUST check that input points properly decompress to points on
+the curve. Removing such checks may result in extremely effective attacks. The
+curve is not twist-secure: implementations using single coordinate ladders MUST
+validate points before operating on them. In the case of protocols that require
+contributory behavior, when the identity is the output of the DH primitive it
+MUST be rejected and failure signaled to higher levels. Notoriously {{?RFC5246}}
+without {{?RFC7627}} is such a protocol.
+
+Implementations MUST ensure that execution of branches and memory addresses
+accessed do not depend on secret data.  The time variability introduced by
+secret-dependent operations have been exploited in the past via timing and cache
+attacks to break implementations.  Side-channel analysis is a constantly moving
+field, and implementers must be extremely careful to ensure that operations do
+not leak any secret information. Using ephemeral private scalars for each
+operation (ideally, limiting the use of each private scalar to one single
+operation) can reduce the impact of side-channel attacks.  However, this might
+not be possible for many applications of Diffie-Hellman key agreement.
+
+In the future quantum computers may render the discrete logarithm problem easy
+on all abelian groups through Shor's algorithm. Data intended to remain
+confidential for significantly extended periods of time SHOULD NOT be protected
+with any primitive based on the hardness of factoring or the discrete log
+problem (elliptic curve or finite field).
 
 --- back
 
 # Constants
 
-ctau1 = 221360928884514619410\*i + 33754435779700894835198039471158097091
```

Comment:
Here's the python script:  https://gist.github.com/bifurcation/9ed3a66fceda5e4d81f019217f2a434c

Now that Patrick has elucidated the true definition of `c` and `c'`, should have code posted later today.